BIGGEST rover commit 😭️

ROMANT21 opened the pull request.

ROMANT21 committed:
BIGGEST rover commit 😭️

onkoe commented:
![image](https://github.com/Sooner-Rover-Team/Autonomous/assets/66580279/6a8a3e3d-955d-4168-b88b-6412bf3951aa)

logbasem commented on main.py:
```diff
@@ -1,122 +1,151 @@
+
+    # Drive to our goal and if we succeed, flash the LEDs green
+    if drive(rover, gps_coordinates, aruco_ids):
+        logger.info("Made it goal! Flash the LEDs green!")
```
Extremely small thing, but missing the word "to" in "Made it to goal!"

onkoe replied:
fixed in 770f335

ROMANT21 committed:
Made some changes lol

onkoe commented:
to get dependencies with Rye, do `rye init --no-import --py 3.10 && rm -rfd src/`, then copy in this `pyproject.toml` file:

```toml
[project]
name = "autonomous"
version = "0.1.0"
description = "Add your description here"
authors = [{ name = "Barrett", email = "contact@barretts.club" }]
dependencies = ["opencv-python>=4.7", "flask>=2.3.2", "loguru>=0.7.2"]
readme = "README.md"
requires-python = ">= 3.10"
[build-system]
requires = ["hatchling"]
build-backend = "hatchling.build"

[tool.rye]
managed = true
dev-dependencies = ["mypy>=1.8.0", "ruff>=0.2.0"]

[tool.hatch.metadata]
allow-direct-references = true

[tool.hatch.build.targets.wheel]
packages = ["./"]

```

and `rye sync` :)

onkoe committed:
chore: Remove YOLO comment

onkoe committed:
fix: Missing `typing` import

onkoe committed:
chore: Remove unused YOLO init

onkoe committed:
fix: `ar.py` Example (so MyPy shuts up)

onkoe committed:
fix: Use modern OpenCV syntax

These lines were errors. Those are from a much older version of `opencv-python` than we're targeting.

onkoe committed:
fix(ARTracker.py): Return a bool in `marker_found`

this needs to return a bool so our callers know what's going on :)

onkoe committed:
docs(ARTracker.py): Add docstrings to some methods

onkoe commented:
lmk if any of these go too far and i'll revert it

tried to keep things minimal though :3

onkoe committed:
docs(ARTracker.py): remove incorrect ARTracker init comments

onkoe committed:
fix(ARTracker): Early return in `marker_found`

accidentally stuck that return before all those calculations.

oopsie :)

onkoe committed:
fix(ARTracker): Remove `use_yolo` arg

...we removed the ARUCO tag YOLO like a year ago lol

onkoe committed:
fix(ARTracker): Give cap some breathing room

otherwise, this can be an infinite loop (as accessing the camera can prevent the system from doing so to further initialize it)

onkoe committed:
feat(ARTracker): Copy image before grayscale

onkoe committed:
chore: Remove unused imports

onkoe committed:
feat(main.py): Improve config loading

onkoe committed:
refactor: Use PEP 8 module names

we had an error that was like:

```python3
import Drive
from Drive import Drive
```

For reasons like this, it's super important to follow the spec for langs that prefer runtime checks...

onkoe committed:
fix: Unused imports in examples

onkoe commented:
CI passes so officially, we've won

onkoe committed:
fix: Correct "goal found" grammar

onkoe committed:
feat: Use Rye

onkoe committed:
fix(examples/ar): Handle case where marker isn't found

onkoe committed:
fix(aruco_tracker.py): Check for detection failure

onkoe committed:
fix(ar_tracker): use Python3.6 (old cv2) syntax

Commit:
fix(main): may 21st debugging (pre-refactor)

Commit:
chore: Default to Python 3.8

Commit:
feat(drive.py): use logger for prints

Commit:
fix(aruco_tracker.py): rename clashing namespace items

Commit:
chore(pyproject.toml): default to Python@3.6

onkoe committed:
fix(ci): small Ruff lints
